pkg/apis/logging/v1/elasticsearch_types.go

```diff
@@ -97,29 +119,70 @@ type ClusterHealth struct {
 // ElasticsearchNode struct represents individual node in Elasticsearch cluster
 // GenUUID will be populated by the operator if not provided
 type ElasticsearchNode struct {
-	Roles          []ElasticsearchNodeRole  `json:"roles"`
-	NodeCount      int32                    `json:"nodeCount"`
-	Resources      v1.ResourceRequirements  `json:"resources"`
-	NodeSelector   map[string]string        `json:"nodeSelector,omitempty"`
-	Tolerations    []v1.Toleration          `json:"tolerations,omitempty"`
-	Storage        ElasticsearchStorageSpec `json:"storage"`
-	GenUUID        *string                  `json:"genUUID,omitempty"`
-	ProxyResources v1.ResourceRequirements  `json:"proxyResources,omitempty"`
+	// The specific Elasticsearch cluster roles the node should perform
+	//
+	// +optional
```

Comment:
not optional

Reply:
Technically correct, but are our former `required` constrains did not include this field. Same question here, what if the API has stored object w/o this field that is translated into an empty slice? Is adding this now to required a breaking change?

Reply:
Actually testing this with a persistent object w/o ManagementState and the new CRD generation has the same effect as previously. It produces error logs on the operator side.

Reply:
that i guess was my point, currently we're doing validation within the operator to make sure we have roles defined (we also make sure we have at least one data and one master... but that's a little different). wouldn't it be easier for us to have this rejected for users when they try to create/update the object instead of a bad change being accepted which they then need to look around to figure out it isn't "valid"